src/Integration/Binding/BindingController.cs

```diff
@@ -0,0 +1,167 @@
+//-----------------------------------------------------------------------
+// <copyright file="BindingController.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.TeamFoundation.Client.CommandTarget;
+using SonarLint.VisualStudio.Integration.Progress;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Integration.WPF;
+using SonarLint.VisualStudio.Progress.Controller;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows.Input;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    /// <summary>
+    /// A dedicated controller for the <see cref="BindCommand"/>
+    /// </summary>
+    internal class BindingController : HostedCommandControllerBase, IBindingWorkflowExecutor
```

Comment:
There's binding controller and connection controller, each of them expose commands which are aggregated at the top level section controller and used for UI databinding. Some of the command could be hidden when they are not bound to the UI (i.e. Don't show again).
The lifecycle of those controller is tied to the section controller, where as the state spans across all the controllers in a VS session and as such defined and managed by other classes that were introduced part of this refactoring (they are the main reason for the refactoring really).